File: app/src/main/java/com/ariefzuhri/myspecificnotification/network/ApiConfig.java
package com.ariefzuhri.myspecificnotification.network;

import static com.ariefzuhri.myspecificnotification.BuildConfig.FCM_BASE_URL;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import android.content.Context;

import androidx.annotation.NonNull;

import com.chuckerteam.chucker.api.ChuckerInterceptor;

public class ApiConfig {

    @NonNull
    public static ApiService getApiService(Context context) {
        ChuckerInterceptor loggingInterceptor = new ChuckerInterceptor.Builder(context)
                .build();

        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(loggingInterceptor)
                .build();

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(FCM_BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .client(client)
                .build();

        return retrofit.create(ApiService.class);
    }
}

File: app/src/main/java/com/ariefzuhri/myspecificnotification/data/network/ApiConfig.kt
package com.ariefzuhri.myspecificnotification.data.network

import com.chuckerteam.chucker.api.ChuckerInterceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import android.content.Context
import com.ariefzuhri.myspecificnotification.BuildConfig

object ApiConfig {

    fun getApiService(context: Context): ApiService {
        val loggingInterceptor = ChuckerInterceptor.Builder(context)
            .build()

        val client = OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .build()

        val retrofit = Retrofit.Builder()
            .baseUrl(BuildConfig.FCM_BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .client(client)
            .build()

        return retrofit.create(ApiService::class.java)
    }
}